### PR TITLE
migrate bgp_instance framework to Plugin Framework

### PR DESCRIPTION
**Pull request by maksym-nazarenko**

### CODE REVIEW

File: docs/resources/bgp_instance.md [hunk 1]
```diff
@@ -26,24 +26,24 @@ resource "mikrotik_bgp_instance" "instance" {
...
-- `routing_table` (String) Name of routing table this BGP instance operates on.  Default: `""`.
+- `ignore_as_path_len` (Boolean) Whether to ignore AS_PATH attribute in BGP route selection algorithm.
+- `out_filter` (String) Output routing filter chain used by all BGP peers belonging to this instance.
+- `redistribute_connected` (Boolean) If enabled, this BGP instance will redistribute the information about connected routes.
+- `redistribute_ospf` (Boolean) If enabled, this BGP instance will redistribute the information about routes learned by OSPF.
+- `redistribute_other_bgp` (Boolean) If enabled, this BGP instance will redistribute the information about routes learned by other BGP instances.
+- `redistribute_rip` (Boolean) If enabled, this BGP instance will redistribute the information about routes learned by RIP.
+- `redistribute_static` (Boolean) If enabled, the router will redistribute the information about static routes added to its routing database.
+- `routing_table` (String) Name of routing table this BGP instance operates on.
```
Comment (ddelnano) [inline docs/resources/bgp_instance.md:42]: Can we update the `Description` string to include the default value? I don't know why tfplugindocs changed that, but I thought it was nice to have those in the docs.

**Review by ddelnano — COMMENTED** [review]
One minor comment about the docs change.

Also it looks like this error assertion is different when it's using the plugin framework
https://github.com/ddelnano/terraform-provider-mikrotik/blob/d3839545ee76e20eb6ebd897bf1829a13c12d223/mikrotik/resource_bgp_instance_test.go#L57

```
=== CONT  TestAccMikrotikBgpInstance_createFailsOnRouterOSv7
    resource_bgp_instance_test.go:50: Step 1/1, expected an error with pattern, no match on: Error running apply: exit status 1
        
        Error: Creation failed
        
          with mikrotik_bgp_instance.bar,
          on terraform_plugin_test.tf line 2, in resource "mikrotik_bgp_instance" "bar":
           2: resource "mikrotik_bgp_instance" "bar" {
        
        from RouterOS device: no such command prefix
--- FAIL: TestAccMikrotikBgpInstance_createFailsOnRouterOSv7 (0.46s)
```

**Comment by maksym-nazarenko** [comment]
>Also it looks like this error assertion is different when it's using the plugin framework

yes, it is not covered by current interface-based implementation of the client (will try to add support for that asap)

File: docs/resources/bgp_instance.md (hunk 1 above)
Comment (maksym-nazarenko) [reply docs/resources/bgp_instance.md:42]: It is [not supported](https://github.com/hashicorp/terraform-plugin-docs/issues/65) by `tfplugindocs` yet.
Related feature was requested in [terraform-plugin-framework](https://github.com/hashicorp/terraform-plugin-framework/issues/625) to have at least workaround for this.
I will take a look - maybe it is possible to contribute to solve this issue 🤷

Comment (maksym-nazarenko) [reply docs/resources/bgp_instance.md:42]: @ddelnano I just found a way to add `Default` to the docs.
We need to implement custom type for each Terraform attribute type (`bool, string, int64, set, list`) like this:
```go
type stringDefault struct {
	schema.StringAttribute
}

func (a stringDefault) GetDescription() string {
	origDescription := a.StringAttribute.GetDescription()
	if a.Default == nil {
		return origDescription
	}

	resp := defaults.StringResponse{}
	a.Default.DefaultString(context.TODO(), defaults.StringRequest{}, &resp)
	origDescription = fmt.Sprintf(`%s Default: %s`, origDescription, resp.PlanValue)

	return origDescription
}
```

and use it as a wrapper in the `Schema()` method (example is based on `mikrotik_bgp_peer` resource):
```go
			"address_families": stringDefault{
				schema.StringAttribute{
					Optional: true,
					Computed: true,
					PlanModifiers: []planmodifier.String{
						stringplanmodifier.UseStateForUnknown(),
					},
					Default:     stringdefault.StaticString("ip"),
					Description: "The list of address families about which this peer will exchange routing information.",
				},
			},
```

Don't look at namings :)

Comment (ddelnano) [reply docs/resources/bgp_instance.md:42]: @maksym-nazarenko nice, that sounds like a great approach until upstream exposes equivalent functionality for the plugin framwork 👍

Comment (maksym-nazarenko) [reply docs/resources/bgp_instance.md:42]: @ddelnano the docs default value issue will be handled in #203 and #204, so this PR can be reviewed on its own.

**Review by ddelnano — APPROVED** [review]

**Pull request merged**